src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1689,7 +1689,7 @@ internal static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSt
                     ImmutableDictionary<Symbol, Symbol> remappedSymbols = null;
                     var compilation = bodyBinder.Compilation;
                     var isSufficientLangVersion = compilation.LanguageVersion >= MessageID.IDS_FeatureNullableReferenceTypes.RequiredVersion();
-                    if (compilation.NullableSemanticAnalysisEnabled)
+                    if (compilation.ShouldRunNullableAnalysis)
```

Comment:
>ShouldRunNullableAnalysis [](start = 36, length = 25)

It is not clear what motivated this change. Given the current implementation and comments for the properties, I cannot tell the difference between the two. If it is important to keep both properties, I think we need to clarify when each of them is supposed to be used. If ```NullableSemanticAnalysisEnabled``` is about SemanticModel``` and ```ShouldRunNullableAnalysis``` is not, then the change doesn't feel appropriate because the comment inside the ```if``` indicates that SemanticModel is involved here in some way.